acme/http.go

```diff
@@ -33,9 +34,11 @@ func httpHead(url string) (resp *http.Response, err error) {
 	req.Header.Set("User-Agent", userAgent())
 
 	resp, err = http.DefaultClient.Do(req)
-	if resp.Body != nil {
-		resp.Body.Close()
+	if err != nil {
+		return resp, err
 	}
+	io.Copy(ioutil.Discard, resp.Body)
```

Comment:
@jehiah Thanks, this is a subtlety I had to look up. The docs for client.Do say:

> Callers should close resp.Body when done reading from it. If resp.Body is not closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

I think that we just have to close it, not necessarily read all of it first. Especially since HEAD requests don't have response bodies.